plugins/connection/incus.py

```diff
@@ -155,11 +155,33 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
         stdout = to_text(stdout)
         stderr = to_text(stderr)
 
-        if stderr == "Error: Instance is not running.\n":
-            raise AnsibleConnectionFailure(f"instance not running: {self._instance()}")
+        if stderr.startswith("Error: ") and stderr.endswith(
+            ": Instance is not running\n"
```

Comment:
How about:
```suggestion
        if stderr.startswith("Error: ") and stderr.rstrip().endswith(
            ": Instance is not running"
```
That way the code doesn't depend on whether there's a newline (or not) at the end.